.github/workflows/minor_version_release.yml

```diff
@@ -25,17 +21,52 @@ jobs:
           python -m pip install --upgrade pip
           pip install pydoc-markdown==4.5.1
 
-      - name: Convert version name to regular semver (e.g. v1.10.x -> v1.10.0)
-        id: regular_semver
-        env:
-          version: ${{github.event.inputs.version}}
+      - name: Define all versions
+        id: versions
+        shell: bash
+        # We only need `major.minor` in Readme so we cut the full version string to the first two tokens
         run: |
-            echo $version | sed -e 's/x$//' | { var=$(cat); echo "VERSION=v${var}0" >> $GITHUB_ENV; }
-            echo $version | sed -e 's/x$//'
+          git fetch
+          git checkout main
+          echo "::set-output name=new_release_version::$(cat VERSION.txt | cut -d "." -f 1,2)"
+          echo "versions.outputs.new_release_version = $(cat VERSION.txt | cut -d "." -f 1,2)"
 
-      - name: Save short version name
-        env:
-          version: $VERSION
+      - name: Create new version branch
         run: |
-          echo $VERSION | sed -e 's/\.[0-9]*$//' | { var=$(cat); echo "VERSION_SHORT=v${var}" >> $GITHUB_ENV; }
-          echo $VERSION | sed -e 's/\.[0-9]*$//'
\ No newline at end of file
+          git checkout -b v${{ steps.versions.outputs.new_release_version }}.x
+          git config --global user.name "github-actions[bot]"
+          git config --global user.email "github-actions[bot]@users.noreply.github.com"
+          git push -u origin v${{ steps.versions.outputs.new_release_version }}.x
+
+      - name: Bump version on main
+        id: bump-version
+        run: |
+          git fetch
+          git checkout main
+          git config --global user.name "github-actions[bot]"
+          git config --global user.email "github-actions[bot]@users.noreply.github.com"
+          git pull
+          cat VERSION.txt | awk -F. '/[0-9]+\./{$2++;print}' OFS=. | tee VERSION.txt
+          cat VERSION.txt
+          echo "::set-output name=new_unstable_version::$(cat VERSION.txt | cut -d "." -f 1,2)"
```

Comment:
Let's call this `next_release_minor` to avoid confusion:
```
echo "::set-output name= next_release_minor::$(cat VERSION.txt | cut -d "." -f 1,2)"
```
We don't seem to use it anywhere BTW

Reply:
I have deleted this line now